gestalt-module/src/main/java/org/terasology/gestalt/naming/SemverExpression.java

```diff
@@ -0,0 +1,32 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.gestalt.naming;
+
+import com.github.zafarkhaja.semver.expr.ExpressionParser;
+import com.google.common.base.MoreObjects;
+
+import java.util.function.Predicate;
+
+public class SemverExpression implements Predicate<Version> {
+
+    private final Predicate<com.github.zafarkhaja.semver.Version> expression;
+    private final String source;
+
+    public SemverExpression(String source) {
+        this.source = source;
+        this.expression = ExpressionParser.newInstance().parse(source);
```

Comment:
TODO: error handling
or maybe using a documented exception here, and error handling in DependencyInfo?